### PR TITLE
[API][Behat] fixed steps for quantity-change api test

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | master 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| License         | MIT

Step was badly implemented, it was adding the items, not changing quantity. It is fixed here.

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Shop/CartContext.php [hunk 1]
```diff
@@ -497,10 +506,10 @@ private function changeQuantityOfOrderItem(string $orderItemId, int $quantity, s
 
     private function hasItemWithNameAndQuantity(Response $response, string $productName, int $quantity): bool
     {
-        $items = json_decode($response->getContent(), true)['hydra:member'];
+        $items = $this->responseChecker->getCollection($response);
 
         foreach ($items as $item) {
-            if ($item['productName'] === $productName && $item['quantity'] === $quantity) {
+            if ($item['quantity'] === $quantity && $item['productName'] === $productName) {
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CartContext.php:512]: Why did you change the order?

File: src/Sylius/Behat/Context/Api/Shop/CartContext.php [hunk 2]
```diff
@@ -319,6 +320,18 @@ public function iShouldSeeWithQuantityInMyCart(string $productName, int $quantit
         $this->checkProductQuantity($this->cartsClient->getLastResponse(), $productName, $quantity);
     }
 
+    /**
+     * @Then I should be informed that cart items are no longer available
+     */
+    public function iShouldBeInformedThatCartItemsAreNoLongerAvailable(): void
+    {
+        $response = $this->cartsClient->getLastResponse() ?? $this->sharedStorage->get('response');
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CartContext.php:328]: Why are you getting a response from shared storage here? If I see it correctly, you are setting this response only after complete step, so it is not associated with the scenarios, that you are changing in this PR

Comment (arti0090) [reply src/Sylius/Behat/Context/Api/Shop/CartContext.php]: some steps are called on CheckoutContext, and there it is using AbstractBrowser class to make requests. I used there a sharedStorage to send objects between contexts. It is needed to make those steps work for now. But soon I will work on refactor of both of these contexts, so it won't be such a confussion.

File: src/Sylius/Behat/Context/Api/Shop/CartContext.php (hunk 1 above)
Comment (arti0090) [reply src/Sylius/Behat/Context/Api/Shop/CartContext.php:514]: Comparing integer is faster 😄  but I can revert this change if it creates confusion 😉

Comment (GSadee) [reply src/Sylius/Behat/Context/Api/Shop/CartContext.php:514]: But there is a greater chance that quantity of different products could be the same and there will be more checks in the result

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, @arti0090! :1st_place_medal:
